### ISSUE TITLE
Mapping browser script folder (srf*) to persistent layer

### ISSUE THREAD

**Issue by SriJaganath** [OracleSiebel/ConfiguringSiebel#87]
Hi Duncan,
Thanks a lot for sharing a great video. I am able to bring up the environment with the step provided.
Have one question, as you aware the browser scripts folder (srf* ) will store under SAI ../webapps/Siebel/ folder.
want to map to this folder to persistent layer. As this folder always create with unique name so how to map to persistent layer?
Is it possible to leverage server.xml file for this as you have done for other custom files ?

Appreciate your response.
Thanks & Regards,
SriJ

**Comment by OracleSiebel** [comment]
Hi SriJ,

Yes, it's possible, but a little more tricky. The latest build system has this feature built in, but I''ve been unable to upload a new version here due to some internal issues. Anyway, I'll try to outline it and you'll have to have a play to see if you can succeed from there.

First here's the code section you can compare with the current initSAI ...

mkdir -p ${sfsPath}/webapps/siebel/files/custom ${sfsPath}/webapps/siebel/images/custom ${sfsPath}/webapps/siebel/scripts/siebel/custom

mkdir -p ${persistencePath}/applicationcontainer/conf/Catalina/localhost
if [ ! -f ${persistencePath}/applicationcontainer/conf/Catalina/localhost/rewrite.config ] ; then
cat <<EOF >> ${persistencePath}/applicationcontainer/conf/Catalina/localhost/rewrite.config
RewriteRule /siebel/srf(.*) /siebel/scripts/siebel/custom/srf$1 [R,NE]
EOF
fi

if ! grep -q "/siebel/images/custom" ${persistencePath}/applicationcontainer/conf/server.xml; then
	sed -i '/unpackWARs="true"/a <Valve className="org.apache.catalina.valves.rewrite.RewriteValve" /><Context docBase="/sfs/webapps/siebel/files/custom" path="/siebel/files/custom" /><Context docBase="/sfs/webapps/siebel/images/custom" path="/siebel/images/custom" /><Context docBase="/sfs/webapps/siebel/scripts/siebel/custom" path="/siebel/scripts/siebel/custom" />' ${persistencePath}/applicationcontainer/conf/server.xml
fi

In a nutshell, the modifies tomcat so that when a request comes in for a folder called srf*****, this then gets mapped to that folder under siebel/custom, which is in turn mapped over to the sfs folder here rather than just the SAI's persistent store. Once we source this from the SFS folder, which is held centrally, it should be easy to direct genbscript to deposit the code there, or you could move it manually.

Hopefully you can unpick this and implement something suitable either by direct copy or by implementing something similar.

Good luck.

**Comment by SriJaganath** [comment]
Thanks for your quick reply. 
let me implement and will post the update.

Thank you

**Comment by OracleSiebel** [comment]
Closing. Note that this solution is built into the latest delivered containers and also into the build system here:

https://github.com/OracleSiebel/ConfiguringSiebel/blob/master/Containerization/Docker/oraclelinux/classic-mde/build/persistence/siebel/standard/initSiebel#L168
